enterprise/views.py

```diff
@@ -585,6 +585,82 @@ def create_enterprise_course_enrollment(request, consent_record, course_id, lice
+        if course_id and self.is_course_run_id(course_id):
+            if license_uuid:
+                enrollment_api_client = EnrollmentApiClient()
+                course_modes = [mode['slug'] for mode in enrollment_api_client.get_course_modes(course_id)]
+                LOGGER.info(
+                    'Retrieved Course Modes for Course {course_id}: {course_modes}'.format(
+                        course_id=course_id,
+                        course_modes=course_modes
+                    )
+                )
+                course_mode = 'verified' if 'verified' in course_modes \
+                    else 'professional' if 'professional' in course_modes \
+                    else 'no-id-professional' if 'no-id-professional' in course_modes \
+                    else 'audit'
+                try:
+                    enrollment_api_client.enroll_user_in_course(
+                        request.user.username,
+                        course_id,
+                        course_mode
+                    )
+                    LOGGER.info(
+                        'Created LMS enrollment for User {user} in Course {course_id} '
+                        'with License {license_uuid} in Course Mode {course_mode}.'.format(
+                            user=request.user.username,
+                            course_id=course_id,
+                            license_uuid=license_uuid,
+                            course_mode=course_mode
+                        )
+                    )
+                except Exception as exc:  # pylint: disable=broad-except
+                    LOGGER.error(
+                        'Unable to create an LMS enrollment from the DSC view: {exc}'.format(
+                            exc=exc
+                        )
+                    )
+                    raise
+            try:
+                self.create_enterprise_course_enrollment(request, enterprise_customer, course_id, license_uuid)
+            except IntegrityError:
+                error_code = 'ENTGDS009'
+                log_message = (
+                    '[Enterprise DSC API] IntegrityError while creating EnterpriseCourseEnrollment.'
+                    'Course: {course_id}, '
+                    'Program: {program_uuid}, '
+                    'EnterpriseCustomer: {enterprise_customer_uuid}, '
+                    'User: {user_id}, '
+                    'License UUID: {license_uuid}, '
+                    'ErrorCode: {error_code}'.format(
+                        course_id=course_id,
+                        program_uuid=program_uuid,
+                        enterprise_customer_uuid=enterprise_customer.uuid,
+                        user_id=request.user.id,
+                        license_uuid=license_uuid,
+                        error_code=error_code,
+                    )
+                )
+                LOGGER.exception(log_message)
+                raise
+            return True
+        return False
```

Comment:
these booleans aren't being used since an exception is raised on failure, should be safe to remove

Reply:
False gets returned if the course id that was provided isn't for a valid course

Reply:
ah, I was looking at the usage within the post() method where the return value doesn't get used. To rephrase: since we're raising an exception for other failures here we should also raise an exception in the case that an invalid course is provided to be consistent

Reply:
Or wait it's something about course_runs I guess? Not exactly sure what that means

Reply:
The course_id needs to be for a course run not a course for that conditional to be true, but that doesn't need to change. What I was saying is that a return value shouldn't be needed from this method. 

After taking a closer look I also realized that the previous behavior didn't fail in any way if that line evaluated to false, it just skipped creating the enrollment. Knowing that, I suggest removing the return statements in `_enroll_leaner_in_course`, which should allow you to simply redirect to the `success_url` in the try block within the `get()` method